shippable.yml

```diff
@@ -1,54 +1,65 @@
+    - poetry run python groclient/lib.py -v
+    # Run tests and code coverage.
+    # http://docs.shippable.com/ci/python-continuous-integration/#test-coverage-reports
     - >
-      pytest
+      mkdir -p shippable/testresults shippable/codecoverage &&
+      poetry run pytest
       --junitxml=shippable/testresults/nosetests.xml
       --cov=groclient
       --cov-report=xml:shippable/codecoverage/coverage.xml
+    # Test sample notebook.
     - >
-      if [ "$SHIPPABLE_PYTHON_VERSION" == "3.7" ]; then
-        shippable_retry pip install -r api/client/samples/analogous_years/requirements.txt && pytest api/client/samples/analogous_years/
-      fi;
+      shippable_retry pip install -r api/client/samples/analogous_years/requirements.txt &&
+      poetry run pytest api/client/samples/analogous_years/
   on_success:
-    # Only build docs on the Python 2.7 run. It is unnecessary to run on both 2.x and 3.x - one or
-    # the other should suffice. However, as of this writing
-    # https://github.com/sphinx-contrib/sphinxcontrib-versioning does not support the latest 3.x
-    # versions, which we do want to run unit tests on. That is why 2.x is used, rather than adding
-    # an older 3.x version to the build matrix just to support docs.
-    #
-    # 2020-10-21 update: sphinxcontrib-versioning is nondeterministically
-    # throwing UnicodeDecode errors, preventing PRs from being merged. While
-    # the library claims to only support up to Python 3.5, it also hasn't been
-    # updated since 2016. It works while manually testing with Python 3.7, so
-    # we're using 3.7 via Shippable to avoid the UnicodeDecode errors.
+    # Build package.
+    # Note: We need to install versioning library outside of the poetry venv.
+    # See note in CONTRIBUTING.md
     - >
-      if [ "$SHIPPABLE_PYTHON_VERSION" == "3.7" ]; then
-        shippable_retry pip install .[docs] &&
-        git config --global user.email "api-documentation@gro-intelligence.com" &&
-        git config --global user.name "Gro Intelligence" &&
-        git remote set-url origin git@github.com:$REPO_FULL_NAME.git &&
-        ssh-agent bash -c 'ssh-add /tmp/ssh/00_sub; sphinx-versioning push -r development docs gh-pages .';
-      fi;
+      pip install poetry-dynamic-versioning &&
+      poetry build
+    # Build docs and push to gh-pages.
+    - >
+      shippable_retry poetry install -E docs &&
+      git config --global user.email "api-documentation@gro-intelligence.com" &&
+      git config --global user.name "Gro Intelligence" &&
+      git remote set-url origin git@github.com:$REPO_FULL_NAME.git &&
+      ssh-agent bash -c 'ssh-add /tmp/ssh/00_sub; poetry run sphinx-versioning push -r development docs gh-pages .'
+    # Publish new development updates to TestPyPI.
+    - >
+      if [ "$BRANCH" == "development" ]; then
+        poetry config repositories.testpypi https://test.pypi.org/legacy/ &&
+        poetry publish -u __token__ -p $TESTPYPI_TOKEN -r testpypi
```

Comment:
it seems shippable is still failing here

Reply:
what's $TESTPYPI_TOKEN? should it be GROAPI_TOKEN?

Reply:
**tokens**

`$TESTPYPI_TOKEN` and `$PYPI_TOKEN` are secrets for publishing to PyPI. They are encoded in the encrypted environment variable also added to our shippable.yml file.

**shippable failure**

The shippable failure is confusing. You can see that shippable actually [succeeds for this branch](https://app.shippable.com/github/gro-intelligence/api-client/runs?branchName=jli-poetry2).

But for some reason, shippable also tries to build in the "context" of the development branch. So the failure is due to [job 3945]( https://app.shippable.com/github/gro-intelligence/api-client/runs/3947/1/console) because that job (for some reason) doesn't include the new environment variables I added.

I tried contacting shippable support, and also filed a [bug report](https://github.com/Shippable/support/issues/5165), but haven't heard back yet.

I *think* that if we merge this PR, that should fix the environment variable issue. If it doesn't, we can remove the shippable config for automatically publishing to PyPI and just do that step manually.

What do you think?